app/src/main/java/com/naxtlevelofandroiddevelopment/marvelgallery/network/MarvelRepository.kt

```diff
@@ -0,0 +1,28 @@
+package com.naxtlevelofandroiddevelopment.marvelgallery.network
+
+import com.naxtlevelofandroiddevelopment.marvelgallery.model.MarvelCharacter
+import com.naxtlevelofandroiddevelopment.marvelgallery.network.providers.Provider
+import com.naxtlevelofandroiddevelopment.marvelgallery.network.providers.RetrofitApi
+import io.reactivex.Single
+
+interface MarvelRepository {
+
+    fun getCharacters(searchQuery: String?): Single<List<MarvelCharacter>>
```

Comment:
I think comment was more like about methods naming, but not about creating generic repository. Spring Data convention is good standard to follow: https://docs.spring.io/spring-data/data-commons/docs/1.6.1.RELEASE/reference/html/repositories.html